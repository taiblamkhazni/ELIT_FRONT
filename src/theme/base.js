/**
 * @file base.js
 * @brief Ce fichier d√©finit le composants de style base.
 */
export const base = {
    breakpoints: ["768px"],
    font: "'Roboto sans', sans-serif",
    backgrounds: {
        light: "#E5E5E5",
        gris: "#F9F9F9",
        white: "#FFFFFF",
    },
    fontWeights: {
        light: "300",
        regular: "400",
        bold: "700",
        black: "900",
    },
    fontSizes: {
        Kilo: "32px",
        Hecto: "24px",
        Deca: "20px",
        Base: "18px",
        Deci: "16px",
        Centi: "14px",
    },
    lineHeights: {
        Kilo: "54px",
        Hecto: "48px",
        Deca: "40px",
        Base: "40px",
        Deci: "32px",
        Centi: "24px",
    },
    colors: {
        primaires: {
            blueLight: "#F7FBFF",
            blue: "#116E9C",
            blueDark: "#1F1A28",
            white: "#FFFFFF",
        },
        secondaires: {
            violet: "#7B83E2",
            vertBlue: "#39B7BF",
            blue: "#6FC1E2",
            orange: "#FFAC5F",
            vert: "#DFE997",
            rose: "#FA927D",
            jaune: "#FFFBA0",
            gris: "#F9F9F9",
            grisMedium: "#CCCCCC",
            grisLight: "#E9E9E9",
            grisDark: "#7A7A7A",
        },
        avertissements: {
            succes: "#10B581",
            danger: "#C91432",
        },
    },
}
