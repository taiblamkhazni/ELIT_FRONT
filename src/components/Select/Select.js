/**
 * @file Select.js
 * @brief Ce fichier d√©finit le composant SelectCustom .
 */
import { Select } from "antd"
import styled from "styled-components"
/**
 * @brief Custom Selector
 */
export const SelectCustom = styled(Select)`
  .antd-select {
    height: 100% !important;
  }
  .ant-select-selector {
    height: 100% !important;
    align-items: center !important;
    font-size: 14px !important;
    border-radius: 4px !important;
    border-color: #CCCCCC !important;
  }
  .ant-select-arrow {
    color: #248BC0 !important;
  }
  .ant-select-selection-search {
    height: 100% !important;
  }
  .ant-select-selection-search-input {
    height: 100% !important;
  }
`
