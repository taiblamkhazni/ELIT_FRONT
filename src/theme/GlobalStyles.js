/**
 * @file GloblaStyles.js
 * @brief Ce fichier définit plusieurs composants de style.
 * @details Chaque composant est un style spécifique.
 */
import { createGlobalStyle, css } from "styled-components"
/**
 *
 * @param theme - The theme to use.
 * @param customColor - The color choosed.
 *
 * @returns Composant.
 */
export const scrollBarStyle = css`
  ::-webkit-scrollbar {
    width: 6px;
    background-color: #f5f5f5;
  }

  ::-webkit-scrollbar-thumb {
    border-radius: 10px;
    -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.1);
    background-color: ${({ theme, customColor }) => customColor ? customColor : theme.colors.primaires.blue};
  }
  scroll-behavior: smooth;
`
/**
 * Determines the color for a project's status button.
 *
 * @param theme - The theme to use.
 *
 * @returns Composant.
 */
export const GlobalStyles = createGlobalStyle`
    body {
        background: ${({ theme }) => theme.backgrounds.white};
        color: ${({ theme }) => theme.colors.primaires.blueDark};
        font-family: ${({ theme }) => theme.font};
        transition: all 0.50s linear;
        box-sizing: border-box;
        margin: 0;
        padding: 0 !important;
        ${scrollBarStyle}
    }
    .ant-modal-wrap {
        overflow-x: hidden;
        ${scrollBarStyle}
    }

    input[type="checkbox"] {
      width: 1.15em;
      height: 1.15em;
      border-radius: 0.15em;
      transform: translateY(-0.075em);
      vertical-align: middle;
      accent-color: ${({ theme }) => theme.colors.primaires.blue};
    }
`
