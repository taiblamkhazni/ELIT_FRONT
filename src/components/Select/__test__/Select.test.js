/**
 * @file Select.test.js
 * @brief Ce fichier contient des tests pour le composant SelectCustom.
 */
import renderer from "react-test-renderer";
/**
 * @brief Import de styled-components.
 */
import { ThemeProvider } from "styled-components";
import { base } from "theme/base";

import { SelectCustom } from "../Select";

/**
 * @brief Import de jest-styled-components.
 */
import "jest-styled-components";
/**
 * @brief Test de l'Ã©chec du composant SelectCustom.
 */
test("SelectCustom component failed", () => {
  const tree = renderer.create(
    <ThemeProvider theme={base}>
      <SelectCustom>SelectCustom</SelectCustom>
    </ThemeProvider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('height', '100% !important', {
    modifier: '.antd-select',
  });
  expect(tree).toHaveStyleRule('height', '100% !important', {
    modifier: '.ant-select-selector',
  });
  expect(tree).toHaveStyleRule('align-items', 'center !important', {
    modifier: '.ant-select-selector',
  });
  expect(tree).toHaveStyleRule('border-radius', '4px !important', {
    modifier: '.ant-select-selector',
  });
  expect(tree).toHaveStyleRule('border-color', '#CCCCCC !important', {
    modifier: '.ant-select-selector',
  });
  expect(tree).toHaveStyleRule('color', '#248BC0 !important', {
    modifier: '.ant-select-arrow',
  });
  expect(tree).toHaveStyleRule('height', '100% !important', {
    modifier: '.ant-select-selection-search',
  });
  expect(tree).toHaveStyleRule('height', '100% !important', {
    modifier: '.ant-select-selection-search-input',
  });
});
