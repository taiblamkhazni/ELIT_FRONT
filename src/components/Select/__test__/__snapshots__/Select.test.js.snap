// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectCustom component failed 1`] = `
.c0 .antd-select {
  height: 100% !important;
}

.c0 .ant-select-selector {
  height: 100% !important;
  -webkit-align-items: center !important;
  -webkit-box-align: center !important;
  -ms-flex-align: center !important;
  align-items: center !important;
  font-size: 14px !important;
  border-radius: 4px !important;
  border-color: #CCCCCC !important;
}

.c0 .ant-select-arrow {
  color: #248BC0 !important;
}

.c0 .ant-select-selection-search {
  height: 100% !important;
}

.c0 .ant-select-selection-search-input {
  height: 100% !important;
}

<div
  className="ant-select ant-select-outlined c0 css-dev-only-do-not-override-1rqnfsa ant-select-single ant-select-show-arrow"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  style={Object {}}
>
  <div
    className="ant-select-selector"
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <span
      className="ant-select-selection-search"
    >
      <input
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autoComplete="off"
        className="ant-select-selection-search-input"
        disabled={false}
        onChange={[Function]}
        onCompositionEnd={[Function]}
        onCompositionStart={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onPaste={[Function]}
        readOnly={true}
        role="combobox"
        style={
          Object {
            "opacity": 0,
          }
        }
        type="search"
        unselectable="on"
        value=""
      />
    </span>
    <span
      className="ant-select-selection-placeholder"
    />
  </div>
  <span
    aria-hidden={true}
    className="ant-select-arrow"
    onMouseDown={[Function]}
    style={
      Object {
        "WebkitUserSelect": "none",
        "userSelect": "none",
      }
    }
    unselectable="on"
  >
    <span
      aria-label="down"
      className="anticon anticon-down ant-select-suffix"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="down"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
        />
      </svg>
    </span>
  </span>
</div>
`;
